controllers/cloud.redhat.com/clowdjobinvocation_controller.go

```diff
@@ -150,19 +164,44 @@ func (r *ClowdJobInvocationReconciler) Reconcile(req ctrl.Request) (ctrl.Result,
 		// We have a match that isn't running and can invoke the job
 		r.Log.Info("Invoking job", "jobinvocation", jobName, "namespace", app.Namespace)
 
-		if err := r.InvokeJob(ctx, &job, &app, &env, &cji); err != nil {
+		if err := r.InvokeJob(&cache, &job, &app, &env, &cji); err != nil {
 			r.Log.Info("Job Invocation Failed", "jobinvocation", jobName, "namespace", app.Namespace)
-			r.Recorder.Eventf(&cji, "Error", "JobNotInvoked", "Job [%s] could not be invoked", jobName)
+			r.Recorder.Eventf(&cji, "Warning", "JobNotInvoked", "Job [%s] could not be invoked", jobName)
 			return ctrl.Result{Requeue: true}, err
 		}
 	}
 
+	if cji.Spec.Iqe.Marker != "" {
+
+		nn := types.NamespacedName{
+			Name:      fmt.Sprintf("%v-%v", app.Name, "iqe"),
+			Namespace: app.Namespace,
+		}
+
+		j := batchv1.Job{}
+
+		r.createIqeJobResource(&cji, &env, &app, nn, ctx, &j)
+		if err := cache.Create(IqeClowdJob, nn, &j); err != nil {
+			r.Log.Info("Iqe Job encountered an error", "jobinvocation", nn.Name, err)
+			return ctrl.Result{Requeue: true}, err
+		}
+
+		cji.Status.Jobs = append(cji.Status.Jobs, j.ObjectMeta.Name)
+		r.Log.Info("Iqe Job Invoked Successfully", "jobinvocation", nn.Name, "namespace", app.Namespace)
+		r.Recorder.Eventf(&cji, "Normal", "ClowdJobInvoked", "Job [%s] was invoked successfully", j.ObjectMeta.Name)
+	}
+
 	// Short running jobs may be done by the time the loop is ranged,
 	// so we update again before the reconcile ends
 	err = r.setCompletedStatus(ctx, &cji)
 	if err != nil {
 		return ctrl.Result{Requeue: true}, err
 	}
+
+	cacheErr := cache.ApplyAll()
```

Comment:
```
	cacheErr := cache.ApplyAll()
	if cacheErr != nil {
		return ctrl.Result{}, err
	}
```

should become for spacesave + readability

```
	if cacheErr := cache.ApplyAll(); cacheErr != nil {
		return ctrl.Result{}, cacheErr
	}
```

also, you are returning `err`, not `cacheErr`